src/dsp_tools/commands/xmlupload/ontology_client.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any, Protocol
+
+from dsp_tools.commands.xmlupload.models.ontology_diagnose_models import Ontology
+from dsp_tools.models.exceptions import BaseError, UserError
+from dsp_tools.utils.connection import Connection
+from dsp_tools.utils.create_logger import get_logger
+from dsp_tools.utils.shared import try_network_action
+
+# pylint: disable=too-few-public-methods
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class OntologyClient(Protocol):
```

Comment:
This protocol is never used to create a `class OntologyClientMock`. There are only instances of `OntologyClientLive`. So I wonder if it is worth having this protocol?